include/ck/tensor_operation/operator_transform/transform_conv_fwd_to_gemm.hpp

```diff
@@ -540,6 +551,559 @@ struct TransformConvFwdToGemm
 
         return out_gemmm_gemmn_desc;
     }
+
+    // Overloaded functions for hipRTC purposes
+    template <typename ALayout,
+              typename std::enable_if<NDimSpatial == 1 &&
+                                          (is_same_v<ALayout, tensor_layout::convolution::G_NW_C> ||
+                                           is_same_v<ALayout, tensor_layout::convolution::NWGC> ||
+                                           is_same_v<ALayout, tensor_layout::convolution::GNWC>),
```

Comment:
Please dont duplicate code of MakeDescriptor functions. Can you just leave function body and call first one from second one?
Something like:

```
    __host__ __device__ static auto
    MakeADescriptor_M_K(const ck::Array<index_t, NDimSpatial + 3>& a_g_n_c_wis_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& a_g_n_c_wis_strides,
                        const ck::Array<index_t, NDimSpatial + 3>& b_g_k_c_xs_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& /* b_g_k_c_xs_strides */,
                        const ck::Array<index_t, NDimSpatial + 3>& c_g_n_k_wos_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& /* c_g_n_k_wos_strides */,
                        const ck::Array<index_t, NDimSpatial>& conv_filter_strides,
                        const ck::Array<index_t, NDimSpatial>& conv_filter_dilations,
                        const ck::Array<index_t, NDimSpatial>& input_left_pads,
                        const ck::Array<index_t, NDimSpatial>& input_right_pads)
    {
     /* Make descriptor logic here */
     }

    template <typename ALayout,
              typename std::enable_if<NDimSpatial == 1 &&
                                          (is_same_v<ALayout, tensor_layout::convolution::G_NW_C> ||
                                           is_same_v<ALayout, tensor_layout::convolution::NWGC> ||
                                           is_same_v<ALayout, tensor_layout::convolution::GNWC>),
                                      bool>::type = false>
    __host__ __device__ static auto
    MakeADescriptor_M_K(const ck::Array<index_t, NDimSpatial + 3>& a_g_n_c_wis_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& a_g_n_c_wis_strides,
                        const ck::Array<index_t, NDimSpatial + 3>& b_g_k_c_xs_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& /* b_g_k_c_xs_strides */,
                        const ck::Array<index_t, NDimSpatial + 3>& c_g_n_k_wos_lengths,
                        const ck::Array<index_t, NDimSpatial + 3>& /* c_g_n_k_wos_strides */,
                        const ck::Array<index_t, NDimSpatial>& conv_filter_strides,
                        const ck::Array<index_t, NDimSpatial>& conv_filter_dilations,
                        const ck::Array<index_t, NDimSpatial>& input_left_pads,
                        const ck::Array<index_t, NDimSpatial>& input_right_pads)
    {
/*Call previous one with these params */ 

}
     
```

Reply:
As I understand it, CK defines 3 versions of the functions based on the number of dimensions, so they all have a different implementation. When I overload the functions, I will need to overload all of the original functions.